source/particle/property/viscoplastic_strain_invariants.cc

```diff
@@ -117,45 +120,11 @@ namespace aspect
             material_inputs.composition[0][i] = solution[this->introspection().component_indices.compositional_fields[i]];
           }
 
-        // Find out plastic yielding by calling function in material model.
-        const MaterialModel::ViscoPlastic<dim> &viscoplastic
-          = Plugins::get_plugin_as_type<const MaterialModel::ViscoPlastic<dim>>(this->get_material_model());
-
-        const bool plastic_yielding = viscoplastic.is_yielding(material_inputs);
-
-
-        /* Next take the integrated strain invariant from the prior time step. When
-         * there are two fields (plastic and viscous), this assumes plastic
-         * strain is always in data position one and viscous strain in position two.
-         * In this case old_strain will first be given the plastic strain, and then,
-         * if there is no plastic yielding it will update to the viscous strain instead.
-         */
-        auto &data = particle->get_properties();
-        double old_strain = data[data_position];
-        if (n_components == 2 && plastic_yielding == false)
-          old_strain = data[data_position+(n_components-1)];
-
-
-        // Calculate strain rate second invariant
-        const double edot_ii = std::sqrt(std::fabs(second_invariant(deviator(material_inputs.strain_rate[0]))));
+        // Evaluate directly in the viscoplastic material model and modify the reaction outputs
+        this->get_material_model().evaluate (material_inputs,material_outputs);
```

Comment:
I like that we now rely on the computation in the material model, which should reduce potential sources of error. However, I remember that this may be way more expensive than the old way to do things (I think the property was written in the current form, because it was much faster this way. 

Do you have a simple model that uses strain healing and that you can compare between the current master branch and your branch? Take a particular look at the "Particles: Update properties" timing section and if that has changed. Dont use old model timing results, because #4060 and #4062 also got merged in the meantime and will have changed particle performance.